vengine/vengine/graphics/UIRenderer.hpp

```diff
@@ -85,27 +90,31 @@ class UIRenderer
     void setBitmapFont(
         const std::vector<std::string>& characters,
         const uint32_t& bitmapFontTextureIndex,
-        const uint32_t& tileWidth,
-        const uint32_t& tileHeight);
+        const glm::vec2& tileDimension);
 
     void setTexture(const uint32_t& textureIndex);
     void renderTexture(
-        const float& x,
-        const float& y,
-        const float& width,
-        const float& height,
-        const uint32_t& u0 = 0,
-        const uint32_t& v0 = 0,
-        const uint32_t& u1 = 1,
-        const uint32_t& v1 = 1
+        const glm::vec2& position,
+        const glm::vec2& dimension,
+        const glm::uvec4 textureCoords = glm::uvec4(0, 0, 1, 1)
+    );
+    void renderTexture(
+        const glm::vec3& position,
+        const glm::vec2& dimension,
+        const glm::uvec4 textureCoords = glm::uvec4(0, 0, 1, 1)
+    );
+    void renderString(
+        const std::string& text,
+        const glm::vec3& position,
```

Comment:
I think it would be more clear to the end user if "position" was called "worldPosition" instead. Same as above, but on another code line.